server/http.go

```diff
@@ -358,7 +358,7 @@ func handleAuthFunc(srv OIDCServer, idpcs []connector.Connector, tpl *template.T
 			if ok {
 				q := url.Values{}
 				q.Set("code", key)
-				ru := httpPathRegister + "?" + q.Encode()
+				ru := strings.TrimRight(baseURL, "/") + "/" + strings.TrimLeft(httpPathRegister, "/") + "?" + q.Encode()
```

Comment:
Actually, it seems weird that you're having to pass the basePath through to so many _different_ places. Is this a thing you only need to do here due to needing to set the `Location` header?